core/templates/dev/head/domain/exploration/QuestionListObjectFactorySpec.js

```diff
@@ -158,6 +158,46 @@ describe('Question List Object Factory', function() {
     ).toBe(true);
   });
 
+  it('should update the question', function() {
+    var newQuestion = QuestionObjectFactory.create(
+      'Fourth State',
+      InteractionObjectFactory.createFromBackendDict({
+        answer_groups: [{
+          correct: false,
+          outcome: {
+            dest: 'Second state',
+            feedback: ['Some feedback'],
+            param_changes: []
+          },
+          rule_specs: [{
+            inputs: {
+              x: 0
+            }
+          }]
+        }],
+        confirmed_unclassified_answers: [],
+        customization_args: {
+          choices: {
+            value: ['Choice 1', 'Choice 2']
+          }
+        },
+        default_outcome: {
+          dest: 'First state',
+          feedback: ['Try again.'],
+          param_changes: []
+        },
+        fallbacks: [],
+        id: 'MultipleChoiceInput'
+      }),
+      'Fifth state content');
+
+    questionList.updateQuestion(3, newQuestion);
+    var returnedQuestionList = questionList.getQuestions();
```

Comment:
questionList.getQuestions() returns an array of Question objects and the returned value is technically not an instance of QustionListFactory. So, I suggest renaming 'returnedQuestionList' to just 'returnedQuestions.